src/weighted_list.h

```diff
@@ -227,7 +230,7 @@ template <typename T> struct weighted_float_list : public weighted_list<double,
     protected:
 
         size_t pick_ent( unsigned int randi ) const override {
-            double picked = static_cast<double>( randi % RAND_MAX ) / static_cast<double>( RAND_MAX ) *
+            double picked = static_cast<double>( randi % UINT_MAX ) / static_cast<double>( UINT_MAX ) *
```

Comment:
This has always been a bit strange, but I think this change makes it stranger.  It doesn't really make sense to reduce an `unsigned int` mod `UINT_MAX`, since that will leave it unchanged except when it's equal to `UINT_MAX`.
```suggestion
            double picked = static_cast<double>( randi ) / static_cast<double>( UINT_MAX ) *
```